apps/ledger-live-desktop/src/renderer/screens/customImage/Step1ChooseImage.tsx

```diff
@@ -1,19 +1,100 @@
-import React from "react";
+import React, { useCallback, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { NFTMetadata } from "@ledgerhq/types-live";
+import { getMetadataMediaTypes } from "~/helpers/nft";
 import { ImageBase64Data } from "~/renderer/components/CustomImage/types";
 import ImportImage from "~/renderer/components/CustomImage/ImportImage";
+import ImportNFT from "~/renderer/components/CustomImage/ImportNFT";
+import NFTGallerySelector from "~/renderer/components/CustomImage/NFTGallerySelector";
 import { StepProps } from "./types";
 import StepContainer from "./StepContainer";
+import { Box } from "@ledgerhq/react-ui";
+import StepFooter from "./StepFooter";
+import {
+  ImageLoadFromNftError,
+  ImageDownloadError,
+} from "@ledgerhq/live-common/customImage/errors";
 
 type Props = StepProps & {
   onResult: (res: ImageBase64Data) => void;
   setLoading: (_: boolean) => void;
 };
 
+const extractNftBase64 = (metadata: NFTMetadata) => {
+  const mediaTypes = metadata ? getMetadataMediaTypes(metadata) : null;
+  const mediaSizeForCustomImage = mediaTypes
+    ? ["original", "big", "preview"].find(size => mediaTypes[size] === "image")
```

Comment:
Can you move the `original`, `big` etc to a variable outside the component or get those from somewhere if it's generic values?